pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -315,6 +408,58 @@ public Double convert(Object value, PinotDataType sourceType) {
     }
   },
 
+  TIMESTAMP {
+    @Override
+    public int toInt(Object value) {
+      throw new UnsupportedOperationException("Cannot convert value from TIMESTAMP to INTEGER");
+    }
+
+    @Override
+    public long toLong(Object value) {
+      return ((Timestamp) value).getTime();
+    }
+
+    @Override
+    public float toFloat(Object value) {
+      throw new UnsupportedOperationException("Cannot convert value from TIMESTAMP to FLOAT");
+    }
+
+    @Override
+    public double toDouble(Object value) {
+      return ((Timestamp) value).getTime();
+    }
+
+    @Override
+    public boolean toBoolean(Object value) {
+      return ((Timestamp) value).getTime() != 0;
```

Comment:
whats the meaning of this?

Reply:
This method will be triggered when the input value is of type `Timestamp`, but the schema defines the field as `BOOLEAN`. In regular cases it should not be used.